x-pack/plugins/actions/server/routes/create.ts

```diff
@@ -46,10 +46,26 @@ export const createActionRoute = (router: IRouter, licenseState: ILicenseState)
       }
       const actionsClient = context.actions.getActionsClient();
       const action = req.body;
-      const actionRes: ActionResult = await actionsClient.create({ action });
-      return res.ok({
-        body: actionRes,
-      });
+      try {
+        const actionRes: ActionResult = await actionsClient.create({ action });
+        return res.ok({
+          body: actionRes,
+        });
+      } catch (e) {
+        if (e instanceof ActionTypeDisabledError) {
+          switch (e.reason) {
+            case 'config':
```

Comment:
so, if the action is disabled because of the whitelist, they'll get a 400, but if it's disabled via license state, they'll get a 403?

I think I would have implemented it so it returned 403 in all cases - but it is true that the 400 can be "fixed" by having the user update their Kibana config, whereas the license is a bit of a different thing to have to update to get "fixed".  In that way, it does nicely provide a different value to help in diagnosis with any problems, and I don't think it's that big of a deal anyway, so 👍.

I guess it's calculated/used in a different place as well, so makes sense that way as well.

I was wondering if we wanted to include the `reason` in the response as well, but I think we'll be fine without it.  We'd want it in there if some other downstream code was trying to determine more precisely why the error occurred, but I can't think of a case of the top of my head who would ever need that.

Reply:
> so, if the action is disabled because of the whitelist, they'll get a 400, but if it's disabled via license state, they'll get a 403?

I will go ahead and change it to 403 for consistency, makes the code cleaner! Prior to this PR, there was a Boom error thrown that turns into a 400.

> I was wondering if we wanted to include the reason in the response as well, but I think we'll be fine without it.

The message will contain the reason but yeah not something that can be looked up programatically easily. We can include it as needed.

Reply:
Done in https://github.com/elastic/kibana/pull/59153/commits/4c387029ce25ae6f5a0764658976d135a523424d